Newspack/Newspack/Stores/AssetStore.swift

```diff
@@ -395,4 +541,315 @@ extension AssetStore {
+    ///
+    func getStoryAssetsNeedingUpload(storyFolder: StoryFolder) -> [StoryAsset] {
+        let context = CoreDataManager.shared.mainContext
+        let fetchRequest = StoryAsset.defaultFetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "folder == %@ AND remoteID == 0 AND type != 'text'", storyFolder)
+        if let results = try? context.fetch(fetchRequest) {
+            return results
+        }
+
+        return [StoryAsset]()
+    }
+
+}
+
+// MARK: - Sync related
+
+extension AssetStore {
+
+    /// Syncs remote assets for the current site's stories.
+    ///
+    /// - Parameter onComplete: A block called after syncing is complete,
+    /// or if there was an error.
+    ///
+    func syncRemoteAssets(onComplete: @escaping (Error?) -> Void) {
+        let folderStore = StoreContainer.shared.folderStore
+        let folders = folderStore.getStoryFoldersWithPosts()
+        let folderUUIDs = folders.map { $0.uuid! }
+        let remoteIDs = getStoryAssetsRemoteIDsForFolders(folders: folders)
+        let dispatchGroup = DispatchGroup()
+        let remote = MediaServiceRemote(wordPressComRestApi: SessionManager.shared.api)
+        var remoteError: Error? = nil
+
+        dispatchGroup.enter()
+
+        remote.fetchMedia(for: remoteIDs) { [weak self] (mediaArray, error) in
+            guard let mediaArray = mediaArray else {
+                remoteError = error
+                LogError(message: "Error fetching remote media: \(error.debugDescription)")
+                dispatchGroup.leave()
+                return
+            }
+
+            // Handle any deleted remote media.
+            let fetchedIDs = mediaArray.map { item -> Int64 in
+                return item.mediaID
+            }
+            let missing = Set(remoteIDs).subtracting(fetchedIDs)
+            if missing.count > 0 {
+                dispatchGroup.enter()
+                self?.handleDeletedRemoteMedia(for: folderUUIDs, remoteIDs: Array(missing)) {
+                    dispatchGroup.leave()
+                }
+            }
+
+            self?.updateAssets(for: folderUUIDs, with: mediaArray, onComplete: {
+                dispatchGroup.leave()
+            })
+        }
+
+        dispatchGroup.notify(queue: .main) {
+            onComplete(remoteError)
+        }
+    }
+
+    /// Pushes any changes to the current site's StoryAssets to the site.
+    ///
+    /// - Parameter onComplete: A block to call when the update is complete.
+    ///
+    func pushUpdatesToRemote(onComplete: @escaping ([Error]) -> Void) {
+        let folderStore = StoreContainer.shared.folderStore
+        let folders = folderStore.getStoryFoldersWithPosts()
+        let assets = getStoryAssetsWithChanges(storyFolders: folders)
+        var remoteErrors = [Error]()
+
+        guard assets.count > 0 else {
+            onComplete(remoteErrors)
+            return
+        }
+
+        let folderUUIDs = folders.map { $0.uuid! }
```

Comment:
Suggestion! (to drop the IUO!)

```suggestion
        let folderUUIDs = folders.compactMap { $0.uuid }
```

Reply:
Oh that's much better.  Thank you!